File: 课程网站/microlectures/1.4-tools/source/first_experiment.py
from pathlib import Path
import numpy as np
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt
fs=100; f=2; N=200; A=1; start=0
n=start+np.arange(N); t=n/fs; x=A*np.sin(2*np.pi*f*t)
E=np.sum(x**2)
print(f'N={x.size} first={x[0]:.12g} last_t={t[-1]:.12g} sum_squares={E:.12g}')
plt.stem(n,x,markerfmt='.');plt.xlabel('n');plt.ylabel('x[n]');plt.grid()
plt.savefig(Path(__file__).resolve().parent/'first-python.png');plt.close()

File: 课程网站/microlectures/1.4-tools/source/first_experiment.m
% 1.4 第一次实验：先预测长度、端点和平方和，再运行。
fs=100; f=2; N=200; A=1; start=0;
n=start+(0:N-1); t=n/fs; x=A*sin(2*pi*f*t);
E=sum(x.^2);
fprintf('N=%d first=%.12g last_t=%.12g sum_squares=%.12g\n',numel(x),x(1),t(end),E);
figure('Visible','off');stem(n,x,'.');xlabel('n');ylabel('x[n]');grid on;
exportgraphics(gcf,fullfile(fileparts(mfilename('fullpath')),'first-matlab.png'));close(gcf);
